python/mxnet/optimizer.py

```diff
@@ -143,8 +143,8 @@ def _update_count(self, index):
         """
         if index not in self._index_update_count:
             self._index_update_count[index] = self.begin_num_update
-        self._index_update_count[index] += 1
-        self.num_update = max(self._index_update_count[index], self.num_update)
```

Comment:
I think this was intended for updating the weights by kvstore? ?  But whatever,  there are still some bugs when training on a single device.